tests/test_core.py

```diff
@@ -165,7 +165,7 @@ def test_default_diagnostic_filename_is_generated_if_no_name_specified(simple_si
 
 def test_default_diagnostic_filename_increments_for_multiple_diagnostics(simple_sim, tmp_path):
     """Test read_diagnostic_from_input method in Simulation class"""
-    simple_sim.read_diagnostics_from_input()
```

Comment:
This test is really just testing code that's in `read_diagnostics_from_input`, so I think it should be called directly. With the changes you've made, the folder "default_output" is no longer created here, so the `shutil.move` line below can be removed, and the `tmp_path` fixture is no longer needed.

Reply:
In fact, one of my goals with the original issue is to make the `shutil.move` "hack" no longer needed. It was used for one other test in this file; that should be changed, too. And then `import shutil` will not be needed.